commerce_coordinator/apps/stripe/pipeline.py

```diff
@@ -231,3 +234,44 @@ def run_filter(self, payment_intent_id=None, **params):
             'payment_intent': payment_intent,
             'redirect_url': receipt_url
         }
+
+
+class RefundPaymentIntent(PipelineStep):
+    """
+    Refunds a payment intent
+    """
+
+    def run_filter(self, payment_intent_id, amount_in_cents, has_been_refunded, **kwargs):  # pylint: disable=arguments-differ
+        """
+        Execute a filter with the signature specified.
+        Arguments:
+            payment_intent_id (str): The Stripe PaymentIntent id to look up.
+            amount_in_cents (decimal): Total amount to refund
+            has_been_refunded (bool): Has this payment been refunded
+            kwargs: arguments passed through from the filter.
+        """
+
+        tag = type(self).__name__
+
+        if not payment_intent_id or not amount_in_cents:
+            logger.info(f'[{tag}] payment_intent_id or amount_in_cents not set, skipping.')
+            return PipelineCommand.CONTINUE.value
+
+        if has_been_refunded:
+            logger.info(f'[{tag}] payment_intent already refunded, skipping.')
+            return {
+                'refund_response': "charge_already_refunded"
+            }
+
+        stripe_api_client = StripeAPIClient()
+
+        try:
+            ret_val = stripe_api_client.refund_payment_intent(
+                payment_intent_id=payment_intent_id,
+                amount=amount_in_cents,
+            )
+            return {
+                'refund_response': ret_val
```

Comment:
According to the [client's](https://github.com/edx/commerce-coordinator/blob/main/commerce_coordinator/apps/stripe/clients.py#L307) return we will only get the id of the refund object back. Could we modify the client return to send the entire refund object back so the next step in the pipeline (which would be creating the CT transaction based on the stripe refund) has access to all the refunds attributes?

Reply:
I dont see why not... ill do that in my work in a moment... since i havent built the stripe tests.